From 273df2423d9226093310cbcaa8b924bb6b5d6586 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 31 May 2021 17:31:33 -0700
Subject: [PATCH] vulkancts:

O_TRUNC value in OperationId enum collides with the macro O_TRUNC
defined in the POSIX header fnctl.h. To avoid the collision undefine
O_TRUNC in this particular sourcefile before its is used in enums
down below.

This is fixed upstream differently
https://github.com/KhronosGroup/VK-GL-CTS/commit/564c6062f72fe7ecd92b4aea1558c441e651c76b

But until we get this module uprev'ed to that, lets use a simpler
workaround

Upstream-Status: Inappropriate [Fixed Differently]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 .../vulkan/spirv_assembly/vktSpvAsmFloatControlsTests.cpp     | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/external/vulkancts/modules/vulkan/spirv_assembly/vktSpvAsmFloatControlsTests.cpp b/external/vulkancts/modules/vulkan/spirv_assembly/vktSpvAsmFloatControlsTests.cpp
index 22e6c75fa..db7f4b54a 100644
--- a/external/vulkancts/modules/vulkan/spirv_assembly/vktSpvAsmFloatControlsTests.cpp
+++ b/external/vulkancts/modules/vulkan/spirv_assembly/vktSpvAsmFloatControlsTests.cpp
@@ -38,6 +38,10 @@
 #include <limits>
 #include <fenv.h>
 
+#ifdef O_TRUNC
+#undef O_TRUNC
+#endif
+
 namespace vkt
 {
 namespace SpirVAssembly
-- 
2.31.1

